module/main.c

```diff
@@ -1033,28 +1038,30 @@ static void handler_HidTimer(s32 data) {
                             flash_read();
                             tele_set_scene(preset_select);
 
-                            for (int i = 0; i < script[INIT_SCRIPT].l; i++)
-                                process(&script[INIT_SCRIPT].c[i]);
+                            run_script(INIT_SCRIPT);
 
-                            for (n = 0; n < 32; n++) input[n] = 0;
+                            for (size_t n = 0; n < 32; n++) input[n] = 0;
                             pos = 0;
                             set_mode(last_mode);
                         }
                         else if (mode == M_TRACK) {
                             if (mod_SH) {
                                 for (int i = 63; i > edit_index + offset_index;
-                                     i--)
-                                    tele_patterns[edit_pattern].v[i] =
-                                        tele_patterns[edit_pattern].v[i - 1];
-                                if (tele_patterns[edit_pattern].l < 63)
-                                    tele_patterns[edit_pattern].l++;
+                                     i--) {
+                                    int16_t v = tele_get_pattern_val(
+                                        edit_pattern, i - 1);
+                                    tele_set_pattern_val(edit_pattern, i, v);
+                                }
+                                uint16_t l = tele_get_pattern_l(edit_pattern);
+                                if (l < 63) {
+                                    tele_set_pattern_l(edit_pattern, l++);
+                                }
                                 r_edit_dirty |= R_ALL;
                             }
                             else {
-                                if (edit_index + offset_index ==
-                                        tele_patterns[edit_pattern].l &&
-                                    tele_patterns[edit_pattern].l < 64) {
-                                    tele_patterns[edit_pattern].l++;
+                                uint16_t l = tele_get_pattern_l(edit_pattern);
+                                if (edit_index + offset_index == l && l < 64) {
```

Comment:
(actually, this PR is probably not a good place, will move this to main repo instead and only post PR specific questions here)